smac/runhistory/runhistory.py

```diff
@@ -88,8 +94,13 @@ def add(self, config, cost, time,
 
         k = RunKey(config_id, instance_id, seed)
         v = RunValue(cost, time, status, additional_info)
-
         self.data[k] = v
+        
+        #also add to fast data structure
+        is_k = InstSeedKey(instance_id,seed)
+        self._configid_to_inst_seed[config_id] = self._configid_to_inst_seed.get(config_id,{})
+        self._configid_to_inst_seed[config_id][is_k] = v
```

Comment:
I needed a bit to figure out what you mean, but I think it should be fixed now. If not, please explain in more depth the issue here.